internal/handlers/pg/pgdb/database_metadata.go

```diff
@@ -218,20 +223,24 @@ func (m *metadata) remove(ctx context.Context) error {
 	return lazyerrors.Error(err)
 }
 
-// formatCollectionName returns collection name in form <shortened_name>_<name_hash>.
-// Changing this logic will break compatibility with existing databases.
-func formatCollectionName(name string) string {
+// collectionNameToTableName returns name in form <shortened_name>_<name_hash>.
+// It replaces special characters with `_`.
+//
+// Deprecated: this function usage is allowed for collection metadata creation only.
+func collectionNameToTableName(name string) string {
 	hash32 := fnv.New32a()
 	must.NotFail(hash32.Write([]byte(name)))
 
+	mangled := specialCharacters.ReplaceAllString(name, "_")
```

Comment:
Do we understand what the probability of having a collision is? The previous approach looked safe - the hashes themselves were pretty unique. If we replace all special characters in two different hashes, is it probable that we will have to equal strings as the result?

Reply:
But hash is built on the `name`, so hash should be different even if the produce name is `__`. 

I just cannot make regular expression to work, could you suggest how I should build regular expression to avoid possible collision?

Reply:
Well, maybe the current approach is already enough? I guess, ideally, for each possible character, we need a different symbol or a different sequence of symbols. What special characters `hash32` can reproduce?

Reply:
hash32 does not produce special characters. It produces hex.

Reply:
Ah! Got it! So, we need this for the prefix only! Then it's unique enough, yes (because we consider the hash  unique enough).

Reply:
Yeah that logic has not change :hugs: